### PR TITLE
Allow Advanced Search by Partial Name

### PR DESCRIPTION
**Pull request by tareq89**
### 🔍 Problem

In the Advanced Search form, the **Search** button remains disabled if any field fails validation.

The `Name` field is used to collect inputs like:

- Child's name
- Mother's name
- Father's name
- Informant's name

Each of these fields is validated using a **Zod schema**, where both `firstname` and `surname` are marked as required. This causes a problem when a user attempts to search using **only `firstname`** or **only `surname`** — the validation fails, marking the form invalid and disabling the Search button.

As a result, valid user actions (like searching by only one part of a name) were unintentionally blocked.

---

### 🛠️ Solution

This PR resolves the issue by **merging the provided name input with a default structure** that includes all required fields set to empty strings:

```ts
const defaultNameFieldValue = {
  firstname: '',
  middlename: '',
  surname: ''
}

const updatedValue = mergeWithoutNullsOrUndefined(
  defaultNameFieldValue,
  actualInputValue
)
```
By doing this, even if a user only types a `firstname`, the other fields (`surname`, `middlename`) will default to `""` instead of `undefined`, avoiding validation errors from the Zod schema.

---

### ✅ Result: User-Friendly Partial Name Search

| Input Scenario              | Before (❌ Search Disabled) | After (✅ Search Enabled) |
|----------------------------|-----------------------------|----------------------------|
| Only `firstname` provided  | ❌                          | ✅                         |
| Only `surname` provided    | ❌                          | ✅                         |
| Both `firstname` & `surname` | ✅                          | ✅                         |

Now, users can search by only one part of a name without facing form validation blocks.

---

### 🧪 Test Coverage

- [x] Partial name inputs (only firstname or surname) no longer throw validation errors.
- [x] Search button gets enabled when either part of the name is filled.
- [x] Full name validation remains intact where both fields are used.

---

### 📎 Related

This change is related to issues introduced after the addition of `Name` fields in advanced search. It fixes a regression where partial input on required fields blocked search functionality.

Country config pull request: https://github.com/opencrvs/opencrvs-countryconfig/pull/848

https://github.com/opencrvs/opencrvs-core/issues/9765

## Checklist

- [x] I have linked the correct Github issue under "Development"
- [x] I have tested the changes locally, and written appropriate tests
- [x] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [x] I have updated the GitHub issue status accordingly

### CODE REVIEW

**Comment by tareq89** [comment]
Had to rename the branch name to make e2e pass.

**Review by cibelius — APPROVED** [review]

File: packages/client/src/v2-events/features/events/Search/TabSearch.tsx [hunk 1]
```diff
@@ -211,11 +211,31 @@ export function TabSearch({
...
+      // when the user tries to search without filling all the NAME fields (ex: { firstname: "Jhon", surname: undefined }).
+
+      if (field.type === FieldType.NAME) {
+        const mergedValue = mergeWithoutNullsOrUndefined(
+          defailtNameFieldValue,
+          value as NameFieldValue
+        )
+        value = mergedValue
+      }
+    }
```
Comment (rikukissa) [inline packages/client/src/v2-events/features/events/Search/TabSearch.tsx]: @tareq89 @cibelius should this be inside the NAME component instead? Can we make it so it never produces invalid values in the first place. My concern here is we are accumulating special rules like this to outside the input component itself. In the future we want to enable countries to define their own input components in their country config and they cannot create these rules outside it
